addons/pos_self_order/static/src/Components/ProductCard/ProductCard.js

```diff
@@ -0,0 +1,57 @@
+/** @odoo-module */
+
+import { Component } from "@odoo/owl";
+import { useSelfOrder } from "@pos_self_order/SelfOrderService";
+import { useForwardRefToParent } from "@web/core/utils/hooks";
+
+export class ProductCard extends Component {
+    static template = "pos_self_order.ProductCard";
+    static props = ["product", "orderLine?", "currentProductCard?", "class?"];
+    static defaultProps = { class: "" };
+    setup() {
+        this.selfOrder = useSelfOrder();
+        useForwardRefToParent("currentProductCard");
+        this.qtyInCart = !this.props.orderLine && this.qtyInCart();
+    }
+    clickOnProduct() {
+        const product = this.props.product;
+        if (!this.canOpenProductMainView(product)) {
+            return;
+        }
+        // if we click on a product from the cart view, it means that
+        //  want to edit it.
+        if (this.selfOrder.page === "/cart") {
+            this.selfOrder.setCurrentlyEditedOrderLine(this.props?.orderLine);
+        }
+        this.selfOrder.setPage("/products/" + product.product_id);
+    }
+
+    canOpenProductMainView(product) {
+        return (
+            this.selfOrder.table ||
+            product.has_image ||
+            product.description_sale ||
+            product.attributes.length
+        );
+    }
+    qtyInCart() {
```

Comment:
this is a little confusing. This class has a method and a field with the same name `qtyInCart`. Maybe rename this method to `getQtyInCart`.